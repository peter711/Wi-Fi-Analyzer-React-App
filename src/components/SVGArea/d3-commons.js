import { scaleLinear } from 'd3';

export function createScale(maxData, maxRange) {
    return scaleLinear()
        .domain([0, maxData])
        .range([0, maxRange]);
}
