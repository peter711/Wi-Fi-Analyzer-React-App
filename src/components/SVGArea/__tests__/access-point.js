import React from 'react';
import { mount, shallow } from 'enzyme';
import { select } from 'd3';
import * as d3Commons from '../d3-commons';

import AccessPoint from '../access-point';

const boxWidth = 300;
const boxHeight = 300;

const props = {
    xScale: d3Commons.createScale(1000, boxWidth),
    yScale: d3Commons.createScale(1000, boxHeight),
    gain: 4,
    frequency: 2.4,
    updateAccessPointCoords: jest.fn()
}

function createSVGElement() {
    const xmlns = "http://www.w3.org/2000/svg";
    const svgElem = document.createElementNS(xmlns, "svg");
    svgElem.setAttributeNS(null, "width", boxWidth);
    svgElem.setAttributeNS(null, "height", boxHeight);
    return svgElem;
}

describe('<AccessPoint/>', () => {
    it('should render', () => {
        const svg = createSVGElement();
        const wrapper = shallow(<AccessPoint
            svg={svg}
            {...props}
        />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should draw 2 circles on SVG', () => {
        const svg = createSVGElement();
        const wrapper = mount(
            <AccessPoint
                svg={svg}
                {...props}
            />
        );

        const d3SVGWrapper = select(svg);
        expect(d3SVGWrapper.selectAll('circle').nodes().length).toBe(2);
    });

    it('should update on scale change', () => {
        const svg = createSVGElement();
        const wrapper = shallow(
            <AccessPoint
                svg={svg}
                {...props}
            />
        );
        const xScale = d3Commons.createScale(1000, 200);
        const yScale = d3Commons.createScale(1000, 200);

        const nextProps = Object.assign(props, { xScale, yScale });
        wrapper.instance().shouldComponentUpdate(nextProps);
    });

});



