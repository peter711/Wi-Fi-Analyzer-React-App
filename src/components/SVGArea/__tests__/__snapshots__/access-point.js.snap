// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccessPoint/> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccessPoint
    frequency={2.4}
    gain={4}
    svg={
      <svg
        height="300"
        width="300"
      >
        <g>
          <circle
            cx="150"
            cy="150"
            fill="rgba(68, 137, 244, 0.4)"
            r="53.01637298881096"
          />
          <circle
            cx="150"
            cy="150"
            fill="grey"
            r="10"
            style="cursor: pointer;"
          />
        </g>
      </svg>
    }
    updateAccessPointCoords={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "radius": 53.01637298881096,
              "x": 150,
              "y": 150,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    xScale={[Function]}
    yScale={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
